src/api/provider.ts

```diff
@@ -5,6 +5,7 @@ export interface APIResponse {
     json?: any
     text: string
     status: number
+    headers: any
```

Comment:
I think we can use Map<string, string> here, see comment below.

Reply:
Header normalization looks good, still need to add the type above to it so other APIProvider implementers and callers know what to expect